EJERCICIOS LECCION 7 - BUCLES FOR-

EJERCICIO 1

1.Escribid un programa que pregunte al usuario una cantidad a invertir, el interés anual y el número de años, 
y muestre por pantalla el capital obtenido en la inversión cada año que dura la inversión.

In [52]:

# for i in lista: >Para cada elemento i en la lista 'lista'

capital = int(input('Dime que cantidad quieres invertir'))
interes = float(input('Dime que cantidad quieres invertir'))
time = int(input('Dime que cantidad quieres invertir'))

#Tengo que obtener la cantidad obtenida por cada ano que dura la inversion 
for ano in range(1,time+1):
  capital_anual = (capital + (capital*interes/100))/time
  print('El capital obtenido en el ano ',ano,' es:', capital_anual)   
  # Tengo que devolver el dato que me solicita y para ello imprimo en cada iteracion del bucle 

El capital obtenido en el ano  1  es: 525.0
El capital obtenido en el ano  2  es: 525.0


EJERCICIO 2

Escribid un programa en el que se pregunte al usuario por una frase y una letra, 
y muestre por pantalla el número de veces que aparece la letra en la frase.

In [53]:


frase= input("Digame una frase")
letra= input("Digame una letra")

letras_encontradas=0

for caracter in frase:
    if letra == caracter:
        letras_encontradas += 1 #letras_encontradas = leras_encontradas + 1

print(letras_encontradas)   # Tengo que devolver el total de letras por lo que printeo una vez que finalice el bucle. 


2


EJERCICIO 3 

Escribid un programa que nos devuelva la tabla de multiplicar de un número especificado en una función inpuT.

In [2]:
#3.Escribid un programa que nos devuelva la tabla de multiplicar de un número especificado en una función input

numero = int(input("Escriba un número"))

for num in range(1,11):
    print(num, "x", numero, "=", num*numero)

1 x 1 = 1
2 x 1 = 2
3 x 1 = 3
4 x 1 = 4
5 x 1 = 5
6 x 1 = 6
7 x 1 = 7
8 x 1 = 8
9 x 1 = 9
10 x 1 = 10


EJERCICIO 4 

4.Dada la siguinte lista de números:
numbers = [12, 75, 150, 180, 145, 525, 50]

Escribid un programa que muestre sólo los números de una lista que cumplan las siguientes condiciones
El número debe ser divisible entre cinco
Si el número es mayor que 150, se salta y se pasa al siguiente número
Si el número es mayor que 500, entonces detenga el bucle


In [38]:


for num in numbers:
    if num%5 ==0:    #Tambien se podria hacer if ultimoDigito(num) == 0 or ultimoDigito(num) ==5:
        print(num)
        if 150 < num < 500:
            print('pass')
            pass
        elif num >500:
            print('break')
            break




75
150
180
pass
145
525
break


EJERCICIO 5 

Dada la siguiente lista:
list1 = [10, 20, 30, 40, 50]
Imprimir la lista en orden inverso

In [9]:

list1 = [10, 20, 30, 40, 50]
print(list1)

list1.sort(reverse=True)
print(list1)


[10, 20, 30, 40, 50]
[50, 40, 30, 20, 10]


EJERCICIO 6 

Dado el siguiente código
for i in ['1','2','3']:
print (i**2)
Usando un try ... except, cread un programa para evitar que nos de un error el código anterior. 
En su lugar, si se encuentra un error, deberéis poner el código necesario para que no nos devuelva error.

In [39]:
# HEMOS DETERMINADO DOS SOLUCIONES, AMBAS CORRECTAS:

# for i in ['1','2','3']:
#     try:                      
#         print (i**2)
#     except:
#         a = [int(i) for i in ['1','2','3']] # a = [1,2,3] 
# for i in a:
#     print (i**2)


for i in ['1','2','3']:
    try:
        print (i**2)
    except:
        print (int(i)**2)



1
4
9
